IdentityCore/src/intune/MSIDIntuneEnrollmentIdsCache.m

```diff
@@ -140,27 +140,32 @@ - (NSString *)enrollmentIdForHomeAccountId:(NSString *)homeAccountId
 }
 
 - (NSString *)enrollmentIdForHomeAccountId:(NSString *)homeAccountId
-                                    legacyUserId:(NSString *)legacyUserId
+                              legacyUserId:(NSString *)legacyUserId
                                    context:(id<MSIDRequestContext>)context
                                      error:(NSError **)error
 {
-    if (homeAccountId)
+    NSString *enrollmentId = nil;
+    
+    // If homeAccountID is provided, try to match by it first.
+    if (homeAccountId != nil)
```

Comment:
nit: no need to check for "!= nil", just if (homeAccountId) will work

Reply:
Right.
It's Authenticator habit of always being as explicit as possible :)